spinalcordtoolbox/reports/assets/_assets/js/main.js

```diff
@@ -215,19 +224,79 @@ function responseHandler(res) {
 }
 
 function set_download_yml_btn_state(marker) {
+  // Return early if marker is null
+  if (marker === null) return;
+
   let disabled = true;
   sct_data.forEach(item => {
       if (item.qc === marker) {
         disabled = false;
       }
   });
+
   if (containsNonLatinCodepoints(marker) === true) {
     // This converts e.g. '\u2718' -> '2718' with corresponding id='download_yaml_btn_2718'
     marker = marker.codePointAt(0).toString(16)
   }
-  document.getElementById("download_yaml_btn_".concat(marker)).disabled = disabled;
+
+  // Safeguard accessing the DOM element
+  let buttonElement = document.getElementById("download_yaml_btn_".concat(marker));
+  if (buttonElement) {
+    buttonElement.disabled = disabled;
+  }
 }
 
 function containsNonLatinCodepoints(s) {
     return /[^\u0000-\u00ff]/.test(s);
 }
+
+function downloadQcStates() {
+  var qcFlags = {};
+  // Fetch all qc flags from local storage
+  sct_data.forEach(function(item, index) {
+      var uniqueId = item.moddate + '_' + item.fname_in + '_' + item.command;
+      var state = localStorage.getItem(uniqueId);
+      if (state) {
+          qcFlags[uniqueId] = state;
+      }
+  });
```

Comment:
One other thing that stands out to me as a (very slight) potential concern is the use of both `sct_data` _and_ `localStorage` as sources of truth during runtime:

- My impression reading the code is that `sct_data` is the "true" record (since the table is based off of `sct_data`, and the table is what the user sees). Whereas `localStorage` is kind of like a "backup" record, and only should be used when reloading the page.
- Yet, the "Save All" download function fetches the values from `localStorage`, rather than `sct_data`. (I think in 99.99% of cases it won't matter, since they'll both have the same values. But, it might be safest for the download function to fetch values from `sct_data` instead, that way the user will always download exactly what they see on the page.)

So, maybe this could be simplified a little by not accessing `localStorage` here, i.e. just accessing the `.qc` property of `item` directly.